lib/localized.dart

```diff
@@ -4,73 +4,85 @@ import 'dart:convert';
-/// The folder containing json files is [assets/i18n] which is generated
-/// by running the script 'flutter pub run localized:main -l [locale codes]'
+/// is responsible for extracting json values from [Localized] strings
+///
+/// The folder containing json files is [assets/i18n] by default.
+/// The localization files are generated by running the script
+/// 'flutter pub run localized:main -l [comma separated locale codes] -d [dir path]'
 ///
 class LocalizationService {
-  /// A [LocalizationsDelegate] delegate
-  /// Creates an instance of the localization delegate
-  /// Requires [locales] as a parameter to create an instance.
+  /// A [LocalizationsDelegate] delegate that creates an instance of this class
+  ///
+  /// [locales] is the list of supported locales
+  ///
+  /// [dirPath] is the path where the JSON files with localizied strings are located,
+  /// it's [assets/i18n] by default.
   ///
   static LocalizationsDelegate<LocalizationService> delegate(
-          List<Locale> locales) =>
-      _LocalizationServiceDelegate(locales);
+          {List<Locale> locales, String dirPath}) =>
+      _LocalizationServiceDelegate(locales: locales, dirPath: dirPath);
 
   static LocalizationService of(BuildContext context) =>
       Localizations.of<LocalizationService>(context, LocalizationService);
 
   final Locale locale;
+  final String dirPath;
 
-  LocalizationService(this.locale);
+  LocalizationService._({this.locale, this.dirPath = 'assets/i18n/'});
 
   Map<String, String> _localizedStrings;
 
   Future<bool> load() async {
     final jsonString =
-        await rootBundle.loadString('assets/i18n/${locale.languageCode}.json');
+        await rootBundle.loadString('$dirPath/${locale.languageCode}.json');
     final Map<String, dynamic> jsonMap = json.decode(jsonString);
     _localizedStrings = jsonMap.map((key, value) {
       return MapEntry(key, value.toString());
     });
     return true;
   }
 
-  /// A function for extracting the value out of JSON file
-  /// Requires [key] as a parameter get the appropriate value.
+  /// Extracts the value out of a JSON file
+  /// Requires [key] as a parameter to get the appropriate value.
   ///
-  String translate(String key) => _localizedStrings[key];
+  String translate(String key) => _localizedStrings[key] ?? key;
 }
 
 /// A [_LocalizationServiceDelegate] delegate extending [LocalizationsDelegate]
 ///
 class _LocalizationServiceDelegate
     extends LocalizationsDelegate<LocalizationService> {
-  const _LocalizationServiceDelegate(this.locales);
+  const _LocalizationServiceDelegate({this.locales, this.dirPath});
 
   /// A list of desired localizations.
   ///
   final List<Locale> locales;
+  final String dirPath;
 
   @override
-  bool isSupported(Locale locale) =>
-      locales.map((e) => e.languageCode).contains(locale.languageCode);
+  bool isSupported(Locale locale) => locales == null
+      ? kMaterialSupportedLanguages.contains(locale.languageCode)
```

Comment:
in GlobalCupertinoLocalizations there is a constant kCupertinoSupportedLanguages. I suppose and higly assume that this will be reuired by iOS part, so I would add it.